File: easy/125.valid-palindrome.py
import string

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        l, r = 0, len(s) - 1 
        while l < r:
            if not str.isalnum(s[l]):
                l += 1
                continue
            if not str.isalnum(s[r]):
                r -= 1
                continue
            if s[l] != s[r]:
                return False
            l += 1
            r -= 1
        
        return True
        

File: easy/125.valid-palindrome.rs
impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let s2:String = s.to_ascii_lowercase().chars().filter(|&x| x.is_ascii_alphanumeric()).collect();
        if s2.len() == 0 {
            return true;
        }
        let mut i:usize = 0;
        let mut j:usize = s2.len()-1;
        let my_vec: Vec<char> = s2.chars().collect();
    while i < j {
      if my_vec[i] != my_vec[j] {
        return false;
      }
      i += 1;
      j -= 1;
    }
    return true;
    }
}